lib/miq_expression/target.rb

```diff
@@ -0,0 +1,63 @@
+class MiqExpression::Target
+  ParseError = Class.new(StandardError)
+
+  def self.parse!(field)
+    parse(field) || raise(ParseError, field)
+  end
+
+  def self.parse(field)
+    match = self::REGEX.match(field) || return
+    model = match[:model_name].classify.safe_constantize || return
+    args = [model, match[:associations].to_s.split(".")]
+    args.push(match[:column]) if match.names.include?('column')
+    args.push(match[:namespace] == self::MANAGED_NAMESPACE) if match.names.include?('namespace')
```

Comment:
having trouble wrapping my head around all of these

moving `REGEX` into `Target` seemed like a good idea.
But this method has too many `if`s